gcc/testsuite/rust/compile/macro-issue1403.rs

```diff
@@ -0,0 +1,31 @@
+// { dg-do compile }
+// { dg-options "-O1 -gdwarf-5 -dA -w" }
+macro_rules! stmt {
+    ($s:stmt) => {
+        $s
+    };
+    ($s:stmt, $($ss:stmt),*) => {
+        $s;
+        stmt!($($ss),*);
+    };
+}
+
+pub fn test() -> i32 {
+    stmt!(
+        let a = 1
+    );
+    stmt!(
+        let b = 2,
+        let c = 3,
+        let d = 4,
+        let e = 5,
+        let f = b + c + d + e
+    );
+    // { dg-final { scan-assembler "14" } }
```

Comment:
I think it might be better to avoid needing dg-options with a specified optimization level here. I would prefer if this was an execution test and verify the result of 14 you could print it and use dg-output to match the print output.

Reply:
ACK to the first part; see #1438 I've just filed.

Generally I suggest to avoid adding more such executions tests -- instead, reduce them ;-) -- for they have considerably more overhead (assemble, link, execute (possibly on a remote target), verify output).

Execution tests will become more useful once we have language support libraries etc. to link against.